blockchain/chain/src/store/chain_store.rs

```diff
@@ -613,6 +658,31 @@ where
     Ok(out)
 }
 
+#[cfg(feature = "json")]
+pub mod headchange_json {
+    use super::*;
+    use blocks::tipset_json::TipsetJson;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Debug, Serialize, Deserialize)]
+    pub enum HeadChangeJson {
+        Current(TipsetJson),
+        Apply(TipsetJson),
+        Revert(TipsetJson),
+    }
+
+    impl From<HeadChange> for HeadChangeJson {
+        fn from(wrapper: HeadChange) -> Self {
+            match wrapper {
+                HeadChange::Current(tipset) => HeadChangeJson::Current((*tipset).clone().into()),
+
+                HeadChange::Apply(tipset) => HeadChangeJson::Apply((*tipset).clone().into()),
+                HeadChange::Revert(tipset) => HeadChangeJson::Revert((*tipset).clone().into()),
+            }
+        }
+    }
+}
```

Comment:
This will not match the Lotus implementation, are you sure you verified against? Enum representations are not internally tagged by default, and the keys are lower case, not first character capitalized.

Also for this, cloning a tipset just to serialize JSON is unnecessary, should take a reference to the tipset in the structure (if that means separating into two if both serialize and deserialize are needed, then it's worth it)